tests/table.spec.ts

```diff
@@ -139,6 +139,176 @@ test.describe('tests complex table operations', () => {
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Less Than
+    await conditionSelect.selectOption('less-than');
+
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page.getByRole('button', { name: 'Conditional' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Less Than Or Equal
+    await conditionSelect.selectOption('less-than-or-equal');
+
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page.getByRole('button', { name: 'Conditional' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Between
+    await conditionSelect.selectOption('is-between');
+    await page.getByPlaceholder('Start value').click();
+    await page.keyboard.type('3');
+    await page.getByPlaceholder('end value').click();
+    await page.keyboard.type('10');
+
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page.getByRole('button', { name: 'Conditional' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Null
+    await conditionSelect.selectOption('is-null');
+
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page.getByRole('button', { name: 'Conditional' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Not Null
+    await conditionSelect.selectOption('is-not-null');
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page.getByRole('button', { name: 'Conditional' }).click();
+    await page.waitForTimeout(delayForGridRender);
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Delete
+    await page.getByRole('button', { name: 'Done' }).click();
+    await page.getByRole('button', { name: 'Delete rule' }).click();
+
+    await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    // Cancel
+    await page.getByRole('button', { name: 'Add New Rule' }).click();
+    await page.locator('#condition-select').first().selectOption('is-not-null');
```

Comment:
Wasn't you that added it, but the div shouldn't have an id `condition-select`. It should use a `shortid` generated id if it needs to use an `id` at all, and we should instead be using a label name or `aria-label` or `data-testid` to identify the element in our tests instead.
`id` should be unique in the whole document: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id